File: 53. Maximum Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        partialSum = 0
        partialSumMin = 0
        res = nums[0]
        for i in range(len(nums)):
            partialSum += nums[i]
            res = max(res, partialSum - partialSumMin)
            partialSumMin = min(partialSum, partialSumMin)
        return res
        

File: 53. Maximum Subarray.cpp
#include <cmath>
using namespace std;

class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int S = 0;
        int maxSub = nums[0];
        int minS = 0;
        for (int num : nums) {
            S += num;
            maxSub = max(maxSub, S - minS);
            minS = min(minS, S);
        }
        return maxSub;
    }
};
